swag/bank.py

```diff
@@ -301,6 +308,226 @@ def set_timezone(self, user, timezone):
+
+            # Check if the value of $wag is correct regardless its propriety
+            if not isinstance(montant, int) or montant < 0:
+                raise InvalidSwagValue
+
+            # Check if the donator have enough $wag:
+            if (
+                self.get_account_info(donator_account_discord_id).swag_balance - montant
+                < 0
+            ):
+                raise NotEnoughSwagInBalance(donator_account_discord_id)
+
+            # Making the donation
+            donator_account.swag_balance -= montant
+            cagnotte.cagnotte_balance += montant
+
+        elif cagnotte.get_info().cagnotte_currency == "$tyle":
+
+            # Check if the value of $tyle is correct regardless its propriety
+            if not isinstance(montant, (int, float, Decimal)) or montant < 0:
+                raise InvalidStyleValue
+
+            # Check if the donator have enough $tyle:
+            if (
+                self.get_account_info(donator_account_discord_id).style_balance
+                - montant
+                < 0
+            ):
+                raise NotEnoughStyleInBalance
+
+            donator_account.style_balance -= montant
+            cagnotte.cagnotte_balance += montant
+
+        # Write donation in the blockchain
+        self.swagdb.blockchain.append(
+            (
+                utcnow().datetime,
+                TransactionType.DONATION,
+                (
+                    donator_account.get_info().id,
+                    cagnotte.get_info().cagnotte_id,
+                    montant,
+                    cagnotte.get_info().cagnotte_currency,
+                ),
+            )
+        )
+
+        if donator_account_discord_id not in cagnotte.get_info().cagnotte_participant:
+            cagnotte.cagnotte_participant.append(
+                donator_account_discord_id
+            )  # TODO voir si ça marche
+
+        self.transactional_save()
+
+    def donner_depuis_cagnotte(
+        self,
+        cagnotte_idx: int,
+        receiver_account_discord_id: int,
+        montant,
+        emiter_account_discord_id: int,
+    ):
+        cagnotte = self.get_active_cagnotte(cagnotte_idx)
+        reciever_account = self.swagdb.get_account(receiver_account_discord_id)
+
+        if emiter_account_discord_id not in cagnotte.get_info().cagnotte_manager:
+            raise NotInGestionnaireGroupCagnotte
+
+        # Check if the €agnotte have enough Money ($wag or $tyle):
+        if cagnotte.get_info().cagnotte_balance - montant < 0:
+            raise NotEnoughMoneyInCagnotte
+
+        # On regarde le type de la cagnotte pour pouvoir correctement choisir les fonctions qui devront être utiliser
+        if cagnotte.get_info().cagnotte_currency == "$wag":
+
+            # Check if the value of $wag is correct regardless its propriety
+            if not isinstance(montant, int) or montant < 0:
+                raise InvalidSwagValue
```

Comment:
It's will be incompatible by the way we use it in swag_client.py : 

```python
            try:
                value = int(
                    "".join(argent for argent in command_style if argent.isnumeric())
                )
            except ValueError:
                raise InvalidSwagValue
```

The here, the exception can't carry the value, because it will not be define if the ValueError exception is raised

Reply:
No but we can carry the string. However, that's outside of the scope of this PR.